corehq/apps/geospatial/forms.py

```diff
@@ -73,13 +81,27 @@ class Meta:
         #       '<a href="{}" target="_blank">support documentation</a>.'),
         #     'https://confluence.dimagi.com/pages/viewpage.action?pageId=164694245'
         # ),
-        choices=GeoConfig.VALID_DISBURSEMENT_ALGORITHMS,
+        choices=DISBURSEMENT_ALGORITHM_OPTIONS,
         required=True,
     )
+    api_token = forms.CharField(
+        label=_("Enter mapbox token"),
+        help_text=_(
+            "Enter your Mapbox API token here. Make sure your token has the correct scope configured"
+            " for use of the Mapbox Matrix API."
+        ),
+        required=False,
+        widget=forms.PasswordInput(),
+    )
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        if toggles.SUPPORT_ROAD_NETWORK_DISBURSEMENT_ALGORITHM.enabled(self.domain):
+            choices = self.fields['selected_disbursement_algorithm'].choices
+            choices.append(self.ROAD_NETWORK_ALGORITHM_OPTION)
+            self.fields['selected_disbursement_algorithm'].choices = choices
```

Comment:
I initially did that but it didn't work. I have to set it explicitly.

If you look at the code around how `choices` are handled, you see the following:
> choices = property(_get_choices, _set_choices)

There's a getter and a setter. The getter returns a deep copy of the choices, so simply appending it won't work, since you're appending to a copy of the actual values and not the values themselves, hence you need to use the setter method.

Reply:
Ah, sneaky! Thanks for the explanation, all good to keep as is then :+1: